cryptographic_estimators/MQEstimator/series/hilbert.py

```diff
@@ -41,27 +40,59 @@ class HilbertSeries(object):
         sage: H = HilbertSeries(10, [2]*15, q=2)
         sage: H
         Hilbert series for system with 10 variables and 15 polynomials over F_2
+
     """
 
     def __init__(self, n: int, degrees: list[int], q=None):
         self._q = q
         self._nvariables = n
         self._degrees = degrees
-        self._ring = PowerSeriesRing(
-            QQ, 'z', default_prec=2*len(self._degrees))
-        z = self._ring.gen()
+        self._prec = 2 * len(self._degrees)
```

Comment:
why twice the degree?

Reply:
The default by PowerSeries is too small. I've noticed that `self._prec = 2 * len(self._degrees)`  should be enough and still fast for most of the cases.

Reply:
Maybe we should document this? In a comment or something.
Because maybe one day the precision will not be enough (maybe because of big numbers). Or is this completely impossible.

Reply:
@Javierverbel What do you think? 

Does the parameters have any correlation that forces `n` to grow at the same pace that `degrees`, or we may end up having problems on big `n`'s (or `q`)?

Reply:
I would add something like  `This precision should work for most of the case since the systems excepted to be evaluated here have a number of variables which is at most the number of equations`

Reply:
Perfect, see my last commit.

Reply:
perfect from my side, can be merged